pkg/codegen/dotnet/gen_program_test.go

```diff
@@ -1,11 +1,96 @@
 package dotnet
 
 import (
+	"os"
+	"path/filepath"
+	"strings"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
 	"github.com/pulumi/pulumi/pkg/v3/codegen/internal/test"
+	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/executable"
 )
 
 func TestGenerateProgram(t *testing.T) {
-	test.TestProgramCodegen(t, "dotnet", GenerateProgram)
+	test.TestProgramCodegen(t,
+		test.ProgramLangConfig{
+			Language:   "dotnet",
+			Extension:  "cs",
+			OutputFile: "MyStack.cs",
+			Check:      checkDotnet,
+			GenProgram: GenerateProgram,
+		},
+	)
+}
+
+func checkDotnet(t *testing.T, path string) {
+	var err error
+	dir := filepath.Dir(path)
+
+	ex, err := executable.FindExecutable("dotnet")
+	assert.NoError(t, err, "Failed to find dotnet executable")
+
+	// We create a new cs-project each time the test is run.
+	projectFile := filepath.Join(dir, filepath.Base(dir)+".csproj")
+	programFile := filepath.Join(dir, "Program.cs")
+	if err = os.Remove(projectFile); !os.IsNotExist(err) {
+		assert.NoError(t, err)
```

Comment:
Nit: `require.NoError` will terminate the test; `assert.NoError` will continue.

```suggestion
		require.NoError(t, err)
```

Reply:
Same feedback throughout wherever errors really ought to terminate the test. TBH, it might be best to have the `Check` functions return an `error` and let the caller `assert` on it.

Reply:
I'll fix the `require/assert`. 

I find the `require`'s clearer then returning errors. Since the compile check is the last check performed, its semantically indistinct. Being able to `assert` is helpful, for example in `defer`ed cleanups.